### ISSUE TITLE
Interaction problem using vmtklevelsetsegmentation

### ISSUE THREAD

**Issue by lxq015** [vmtk/vmtk#434]
Hello, I tried to follow along the Level Set Segmentation Tutorials. But I have encountered some problems and want to ask for your help.

Every time I get to a step where I need to enter a lower threshold, the interactive window starts blinking and the terminal keeps popping up "Please input lower threshold ('n' for none):". 

Although I entered the corresponding value on the keyboard, this phenomenon kept repeating, so that I had to close the interactive window.

Could you please give me some advice?

ubuntu 18.04
vmtk 1.5.0

**Comment by lxq015** [comment]
The content displayed on the terminal is as follows:
......
Please choose initialization type: 
0: colliding fronts;
1: multiple colliding fronts;
2: fast marching;
3: threshold;
4: isosurface;
5: seed;
Multiple colliding fronts initialization.
Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): Please input lower threshold ('n' for none): .......

**Comment by francesco-ballarin** [comment]
Hi,
I am affected by the same issue as well.

Operating system is debian testing, with python 3.10. `vmtk` was compiled from source, with superbuild on, as of current master.

The issue might be in `pypeScript.InputText`: it seems that the first call to `self.InputStream.prompt(prompt)` waits for the user to press enter, while any successive call to `self.InputStream.prompt(prompt)` (or to `self.InputStream.prompt(prompt,info=False)` in case of validation failure) returns an empty string.

**Comment by carlos-ar** [comment]
Hi, same issue here. I've tried installing with conda-forge both the vmtk-1.5 version with
`Package name: numpy           - Version: 1.21.6    
Package name: vtkmodules.all  - Version: 9.1.0     
Package name: itk             - Version: 5.2.0     
`
and compiling from source with the `vmtk=1.4` branch all using `Python=3.7` on ubuntu 20.04. 

I've made both a script creating the `vmtkLevelsetsegmentation()` class and running the script using the pypes interface. I think this is a tkinter issue that seems to be pervasive in other guis. I do not know enough to troubleshoot this though. Any guidance would be appreciated. Might try to compile vmtk 1.4 by rolling back to an older python version just to get things working for now though.
https://github.com/The-Compiler/pytest-xvfb/issues/11

**Comment by lassoan** [comment]
I'm not sure if anyone still uses pypes scripts to interact with the data (provide inputs, visualization, etc.).

Instead, for interactive operations, most people now use [3D Slicer](https://www.slicer.org). If you install the VMTK extension then you get VMTK and also a number of modules with convenient GUI for vessel segmentation, centerline extraction, metric computations, and lots of other data analysis, quantification, and visualization features. You can also customize and extend these modules in Python to do exactly what you need.

**Comment by carlos-ar** [comment]
I agree, I am using 3D slicer to get ROI's in order to then run a low-level (marching cubes or levelset) segmentation script. I am in the early stages of using the VMTK extension to do multiple vessel segmentations, then merging them into one larger vessel tree. However, I am alternating between writing a python script that uses the  vmtkscripts and doing this interactively in 3D slicer, and getting stuck at different points.

**Comment by marinocj** [comment]
Is there currently a known way to fix this issue or is using VMTK within 3D slicer the best option. I am currently encountering this same error and am wondering if there are any know fixes I could attempt. Any help is greatly appreciated!

**Comment by lassoan** [comment]
I can help with the followings:
- If you have any questions about how to use VMTK filters in 3D Slicer, including any interactions (such as specify input points) then I or other community members can answer. Your question is seen by more people and likely get answered sooner if you post it at https://discourse.slicer.org/c/community/vmtk/28
- If you have any fixes for pure VTK-based interactors in VMTK and you send a pull-request then I can merge it.

**Comment by lassoan** [comment]
Windows, linux, macOS wheels are now published on PyPI for 1.5.0.
